### PR TITLE
Flp 72 earnings report

### PR DESCRIPTION
**Pull request by JackWorkAccount**

### CODE REVIEW

File: src/DataAccess/Repositories/EarningsQueryRepository.cs [hunk 1]
```diff
@@ -30,7 +30,8 @@ public async Task<ProviderEarningsSummary> GetProviderSummary(long ukprn, short
             {
                 TotalLevyEarningsForCurrentAcademicYear = await DbContext.Earning.Where(x => x.UKPRN == ukprn && x.AcademicYear == academicYear && x.FundingType == FundingType.Levy).SumAsync(x => x.Amount),
                 TotalNonLevyEarningsForCurrentAcademicYear = await DbContext.Earning.Where(x => x.UKPRN == ukprn && x.AcademicYear == academicYear && x.FundingType == FundingType.NonLevy).SumAsync(x => x.Amount),
-            };
+                Ukprn =  DbContext.Earning.Where(x => x.UKPRN == ukprn && x.AcademicYear == academicYear).Select(x => x.UKPRN).FirstOrDefault(),
```
Comment (MichaelYoung1981) [inline src/DataAccess/Repositories/EarningsQueryRepository.cs:33]: This doesn't make sense, why not just use the value passed in?  Although given it's an input, it doesn't need to be an output.

File: src/InnerApi/Controllers/ProviderEarningsController.cs [hunk 2]
```diff
@@ -26,9 +26,9 @@ public async Task<IActionResult> Summary(long ukprn)
             return Ok(response.ProviderEarningsSummary);
         }
 
-        [Route("")]
+        [Route("GenerateCSV")]
         [HttpGet]
-        public async Task<IActionResult> Detail(long ukprn)
+        public async Task<IActionResult> GenerateCSV(long ukprn)
```
Comment (MichaelYoung1981) [inline src/InnerApi/Controllers/ProviderEarningsController.cs:31]: The inner api doesn't care that it's generating a csv, that's a FE concern.

File: src/DataAccess/Repositories/EarningsQueryRepository.cs [hunk 3]
```diff
@@ -24,13 +24,13 @@ public async Task Add(Apprenticeship apprenticeship)
...
         {
             var dbResponse = new
             {
-                levyEarnings = await DbContext.Earning.Where(x => x.UKPRN == ukprn && x.AcademicYear == currentAcademicYear && x.FundingType == FundingType.Levy).SumAsync(x => x.Amount),
-                nonLevyEarnings = await DbContext.Earning.Where(x => x.UKPRN == ukprn && x.AcademicYear == currentAcademicYear && x.FundingType == FundingType.NonLevy).SumAsync(x => x.Amount),
-                transferEarnings = await DbContext.Earning.Where(x => x.UKPRN == ukprn && x.AcademicYear == currentAcademicYear && x.FundingType == FundingType.Transfer).SumAsync(x => x.Amount)
+                levyEarnings = await DbContext.Earning.Where(x => x.UKPRN == ukprn && x.AcademicYear == academicYear && x.FundingType == FundingType.Levy).SumAsync(x => x.Amount),
+                nonLevyEarnings = await DbContext.Earning.Where(x => x.UKPRN == ukprn && x.AcademicYear == academicYear && x.FundingType == FundingType.NonLevy).SumAsync(x => x.Amount),
+                transferEarnings = await DbContext.Earning.Where(x => x.UKPRN == ukprn && x.AcademicYear == academicYear && x.FundingType == FundingType.Transfer).SumAsync(x => x.Amount)
```
Comment (tom-gough) [inline src/DataAccess/Repositories/EarningsQueryRepository.cs:33]: will we have FundingType data at this point?

Comment (tom-gough) [reply src/DataAccess/Repositories/EarningsQueryRepository.cs:33]: ok makes sense

**Review by tom-gough — APPROVED** [review]

**Pull request merged**